compiler/luci/pass/src/QuantizationUtils.cpp

```diff
@@ -99,6 +99,13 @@ void compute_asym_scale_zp(float min, float max, float &scaling_factor, int64_t
     nudged_zero_point = static_cast<uint8_t>(std::round(zero_point_double));
   }
 
+  // protect scale from being very low due to overflow
+  if (scale < 1e-5)
+  {
+    scale = 1e-5;
+    nudged_zero_point = static_cast<uint8_t>(std::round(qmin_double - rmin / scale));
```

Comment:
(optional)

I had nasty bug before due to undefined behavior of `static_cast<uint*_t>(float_value)` when `float_value` is out of scope of destination type.

c++'s implicit conversion rule for float->integral:
> If the value cannot fit into the destination type, the behavior is undefined (even when the destination type is unsigned, modulo arithmetic does not apply).

_(from [cppreference.com](https://en.cppreference.com/w/cpp/language/implicit_conversion#Floating.E2.80.93integral_conversions))_

I strongly recommend to make a safer wrapper like:
```c++
uint8_t my_cast(float f)
{
  assert(std::numeric_limits<uint8_t>::min() <= f);
  assert(f <= std::numeric_limits<uint8_t>::max());
  return static_cast<uint8_t>(f);
}
```

Reply:
I think @cgbahk 's comment is worth applying.